frontend/src/framework/internal/components/SelectEnsemblesDialog/selectEnsemblesDialog.tsx

```diff
@@ -17,15 +17,41 @@ import { Switch } from "@lib/components/Switch";
 import { TableSelect, TableSelectOption } from "@lib/components/TableSelect";
 import { useValidState } from "@lib/hooks/useValidState";
 import { ColorSet } from "@lib/utils/ColorSet";
-import { Add, Check, Remove } from "@mui/icons-material";
+import { resolveClassNames } from "@lib/utils/resolveClassNames";
+import { Add, Check, Info, Remove } from "@mui/icons-material";
 import { useQuery } from "@tanstack/react-query";
 
 import { isEqual } from "lodash";
+import { v4 } from "uuid";
 
 import { UserAvatar } from "./private-components/userAvatar";
 
 import { LoadingOverlay } from "../LoadingOverlay";
 
+const CASE_UUID_ENSEMBLE_NAME_SEPARATOR = "~@@~";
+
+// Ensemble element item for a delta ensemble
+export type DeltaEnsembleElementItem = {
+    caseUuid: string;
+    ensembleName: string;
+};
+
+// Internal type before applying created delta ensemble externally
+type InternalDeltaEnsembleItem = {
+    compareEnsemble: DeltaEnsembleElementItem | null; // Allows null
+    referenceEnsemble: DeltaEnsembleElementItem | null; // Allows null
+    uuid: string; // To allow for equal compare and reference ensembles during editing in the dialog
+    color: string;
+    customName: string | null;
+};
+
+export type DeltaEnsembleItem = {
+    compareEnsemble: DeltaEnsembleElementItem;
+    referenceEnsemble: DeltaEnsembleElementItem;
+    color: string;
+    customName: string | null;
+};
+
```

Comment:
I think the naming and the types could be improved by e.g. having a base ensemble reference type being used by both regular and delta ensemble. However, this is going to be subject to some larger refactoring anyways. Hence, we can leave it as is and look at it later. I am not looking too much at this file. As long as everything works, it's fine for now. Let's talk about how we would like to have it/change it when refactoring.